### PR TITLE
Support sending rewards to a reserve

### PR DESCRIPTION
**Pull request by jophish**

### CODE REVIEW

File: contracts/farm-bot.sol [hunk 1]
```diff
@@ -202,46 +206,49 @@ contract FarmBot is Owned, FarmbotERC20 {
...
+		_interestAccrued[i] = TokenAmount(rewardsTokens[i], _tokenBalances[i]);
+	    }
+
+	    TokenAmount[] memory _bountyFees = revoBounty.calculateBountyFee(_interestAccrued);
+	    TokenAmount[] memory _reserveFees = revoBounty.calculateReserveFee(_interestAccrued);
+	    require(_bountyFees.length == _reserveFees.length, "Got conflicting results from RevoBounty");
+	    for (uint i=0; i < _bountyFees.length; i++) {
+		_bountyAmounts[i] = _bountyFees[i].amount;
+		_reserveAmounts[i] = _reserveFees[i].amount;
 		require(_bountyAmounts[i] <= maxFeeNumerator * _tokenBalances[i] / maxFeeDenominator, "Bounty amount too high");
```
Comment (cajubelt) [inline contracts/farm-bot.sol:225]:
```suggestion
		require(_bountyAmounts[i] + _reserveAmounts[i] <= maxFeeNumerator * _tokenBalances[i] / maxFeeDenominator, "Bounty amount too high");
```

Comment (cajubelt) [reply contracts/farm-bot.sol:225]: The goal of this check is to make sure the user has a guaranteed fee maximum that the owners can't change once the contract is deployed. We should include the reserve fee in this guarantee.

**Review by cajubelt — APPROVED** [review]
looks good but see comment

File: contracts/farm-bot.sol (hunk 1 above)
Comment (jophish) [reply contracts/farm-bot.sol:225]: Ahh of course; that's a great catch, thanks!

**Pull request merged**